app/src/main/java/com/example/inharmony/ChatAdapter.java

```diff
@@ -0,0 +1,204 @@
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return isMe;
+    }
+
+    @NonNull
+    @Override
+    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        if (viewType == MESSAGE_INCOMING) {
+            View contactView = inflater.inflate(R.layout.message_incoming, parent, false);
+            return new IncomingMessageViewHolder(contactView);
+        } else if (viewType == MESSAGE_OUTGOING) {
+            View contactView = inflater.inflate(R.layout.message_outgoing, parent, false);
+            return new OutgoingMessageViewHolder(contactView);
+        } else {
+            throw new IllegalArgumentException("Unknown view type");
+        }
+    }
+
+    public abstract class MessageViewHolder extends RecyclerView.ViewHolder {
+        public MessageViewHolder(@NonNull View itemView) {
+            super(itemView);
+        }
+
+        abstract void bindMessage(Message message);
+    }
+
+    public class IncomingMessageViewHolder extends MessageViewHolder {
+        ImageView imageOther;
+        TextView body;
+        TextView name;
+        TextView tvIncomingTrack;
+        ImageView ivIncomingTrack;
+        ImageView ivPlayIncoming;
+
+          public IncomingMessageViewHolder(View itemView) {
+            super(itemView);
+            imageOther = itemView.findViewById(R.id.ivProfileOther);
+            body = itemView.findViewById(R.id.tvOtherBody);
+            name = itemView.findViewById(R.id.tvSender);
+            tvIncomingTrack = itemView.findViewById(R.id.tvIncomingTrack);
+            ivIncomingTrack = itemView.findViewById(R.id.ivIncomingTrack);
+            ivPlayIncoming = itemView.findViewById(R.id.ivPlayIncoming);
+        }
+
+        @Override
+        void bindMessage(Message message) {
+            ParseUser user = message.getSender();
+            try {
+                name.setText(user.fetch().getUsername());
+                if (user.get("profilePic") != null) {
+                    ParseFile image = (ParseFile) user.get("profilePic");
+                    Glide.with(mContext).load(image.getUrl()).into(imageOther);
+                }
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+
+            if (message.getBody() == null) {
+                body.setVisibility(View.GONE);
+            } else {
+                body.setVisibility(View.VISIBLE);
+                body.setText(message.getBody());
+            }
+
+            if (message.getTrackId() == null) {
+                tvIncomingTrack.setVisibility(View.GONE);
+                ivPlayIncoming.setVisibility(View.GONE);
+                ivIncomingTrack.setVisibility(View.GONE);
```

Comment:
I see a few instance of assigning `tvIncomingTrack`, `ivPlayIncoming`, and `ivIncomingTrack` to the same value -- consider creating a util function.